plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java

```diff
@@ -1653,6 +1658,21 @@ public void close()
             }
         }
 
+        private void ensureExtraBytesCapacity(int extraBytesCapacity)
+        {
+            int totalBytesCapacity = bufferSize + extraBytesCapacity;
+            if (buffer.length < totalBytesCapacity && buffer.length < partSize) {
```

Comment:
See https://github.com/trinodb/trino/pull/14212/commits/8408da0af9aaf29df359e1d719a3bd6ea1970343#r986804499. Then you can add assertion here:
```
checkArgument(totalBytesCapacity <= partSize)
```
and then you also dont need:
```
buffer.length < partSize
```